homeassistant/components/meteo_france/config_flow.py

```diff
@@ -40,23 +49,81 @@ async def async_step_user(self, user_input=None):
             return self._show_setup_form(user_input, errors)
 
         city = user_input[CONF_CITY]  # Might be a city name or a postal code
-        city_name = None
+        latitude = user_input.get(CONF_LATITUDE)
+        longitude = user_input.get(CONF_LONGITUDE)
 
-        try:
-            client = await self.hass.async_add_executor_job(meteofranceClient, city)
-            city_name = client.get_data()["name"]
-        except meteofranceError as exp:
-            _LOGGER.error(
-                "Unexpected error when creating the meteofrance proxy: %s", exp
-            )
-            return self.async_abort(reason="unknown")
+        if not latitude:
+            client = MeteoFranceClient()
+            places = await self.hass.async_add_executor_job(client.search_places, city)
+            _LOGGER.debug("places search result: %s", places)
+            return await self.async_step_cities(places=places)
 
         # Check if already configured
-        await self.async_set_unique_id(city_name)
+        await self.async_set_unique_id(f"{latitude}, {longitude}")
         self._abort_if_unique_id_configured()
 
-        return self.async_create_entry(title=city_name, data={CONF_CITY: city})
+        return self.async_create_entry(
+            title=city, data={CONF_LATITUDE: latitude, CONF_LONGITUDE: longitude},
+        )
 
     async def async_step_import(self, user_input):
         """Import a config entry."""
         return await self.async_step_user(user_input)
+
+    async def async_step_cities(self, user_input=None, places=None):
+        """Step where the user choose the city from the API search results."""
+        if places and len(places) > 1 and self.source != SOURCE_IMPORT:
+            places_for_form = {}
+            for place in places:
+                places_for_form[_build_place_key(place)] = f"{place}"
+
+            return await self._show_cities_form(places_for_form)
+        if places and not user_input:
```

Comment:
Add a comment for import and 1 element cases

Reply:
OK